changelog/upgrade-notes.md

```diff
@@ -2,6 +2,26 @@
 
 ### Next release
 
+#### Removal of ol.FeatureStyleFunction
+
+The signature of the vector style function passed to the feature has changed. The function now always take the `feature` and the `resolution` as aguments, the `feature` is no longer bound to `this`.
+
+Old code:
+```js
+style = function(resolution) {
+  var text = this.get('name');
+  ...
+}
+```
+
+New code:
+```js
+style = function(feature, resolution) {
+  var text = feature.get('name');
+  ...
+}
```

Comment:
It might be worth using `feature.setStyle()` to make it clearer where this belongs.